lib/lightning/accounts.ex

```diff
@@ -492,44 +494,55 @@ defmodule Lightning.Accounts do
 
   ## Examples
 
-      iex> deliver_update_email_instructions(user, current_email, &Routes.user_update_email_url(conn, :edit, &1))
+      iex> deliver_update_email_instructions(user, new_email, &Routes.user_update_email_url(conn, :edit, &1))
       {:ok, %{to: ..., body: ...}}
 
   """
   def deliver_update_email_instructions(
         %User{} = user,
-        current_email,
+        new_email,
         update_email_url_fun
       )
       when is_function(update_email_url_fun, 1) do
     {encoded_token, user_token} =
-      UserToken.build_email_token(user, "change:#{user.email}", current_email)
+      UserToken.build_email_token(user, "change:#{user.email}", new_email)
 
     Repo.insert!(user_token)
 
     UserNotifier.deliver_update_email_warning(
       user,
-      current_email
+      new_email
     )
 
     UserNotifier.deliver_update_email_instructions(
-      user,
+      %User{
+        email: new_email,
+        first_name: user.first_name,
+        last_name: user.last_name
+      },
       update_email_url_fun.(encoded_token)
     )
   end
 
-  def validate_change_user_email(user, params \\ %{}) do
+  def validate_change_user_email(user, params \\ %{}, opts \\ []) do
+    validate_password? = Keyword.get(opts, :validate_password, false)
+
     data = %{email: nil, current_password: nil}
     types = %{email: :string, current_password: :string}
 
     {data, types}
-    |> Ecto.Changeset.cast(params, Map.keys(types))
-    |> Ecto.Changeset.validate_required([:email, :current_password])
-    |> Ecto.Changeset.validate_format(:email, ~r/^[^\s]+@[^\s]+$/,
+    |> Changeset.cast(params, Map.keys(types))
+    |> Changeset.validate_required([:email])
+    |> Changeset.validate_format(:email, ~r/^[^\s]+@[^\s]+$/,
       message: "must have the @ sign and no spaces"
     )
-    |> Ecto.Changeset.validate_length(:email, max: 160)
-    |> Ecto.Changeset.validate_change(:email, fn :email, email ->
+    |> Changeset.validate_length(:email, max: 160)
+    |> validate_email_change(user)
```

Comment:
We already have functions in User that should validate the email so it looks like parts of this function and `validate_email_change/2` are redundant. It's been this way for a while, could you use a function inside `User` instead and ensure the behaviour is the same.

Reply:
This is done, thanks